validator/client/runner.go

```diff
@@ -39,41 +36,22 @@ func run(ctx context.Context, v iface.Validator) {
 	cleanup := v.Done
 	defer cleanup()
 
-	headSlot, err := initializeValidatorAndGetHeadSlot(ctx, v)
-	if err != nil {
-		return // Exit if context is canceled.
-	}
-	if err := v.UpdateDuties(ctx, headSlot); err != nil {
-		handleAssignmentError(err, headSlot)
-	}
-	eventsChan := make(chan *event.Event, 1)
-	healthTracker := v.HealthTracker()
-	runHealthCheckRoutine(ctx, v, eventsChan)
-
-	accountsChangedChan := make(chan [][fieldparams.BLSPubkeyLength]byte, 1)
-	km, err := v.Keymanager()
-	if err != nil {
-		log.WithError(err).Fatal("Could not get keymanager")
-	}
-	sub := km.SubscribeAccountChanges(accountsChangedChan)
-	// check if proposer settings is still nil
-	// Set properties on the beacon node like the fee recipient for validators that are being used & active.
-	if v.ProposerSettings() == nil {
-		log.Warn("Validator client started without proposer settings such as fee recipient" +
-			" and will continue to use settings provided in the beacon node.")
-	}
-	if err := v.PushProposerSettings(ctx, km, headSlot, true); err != nil {
-		log.WithError(err).Fatal("Failed to update proposer settings")
+	if err := v.Init(ctx); err != nil {
+		if errors.Is(err, context.Canceled) {
+			return // Exit if context is canceled.
+		}
+		log.WithError(err).Fatal("Failed to initialize validator")
```

Comment:
```suggestion
		log.WithError(err).Error("Failed to initialize validator")
```
Fatal will abruptly stop the node and we return immediately anyway